aten/src/ATen/native/cuda/UnaryGeometricKernels.cu

```diff
@@ -72,6 +72,25 @@ void sin_kernel_cuda(TensorIterator& iter) {
   });
 }
 
+template<typename scalar_t>
+__host__ __device__ static inline scalar_t cos_wrapper(scalar_t v) {
+  return ::cos(v);
+}
+
+template<typename T>
+__host__ __device__ static inline thrust::complex<T> cos_wrapper(thrust::complex<T> v) {
+  return thrust::cos(v);
+}
+
+void cos_kernel_cuda(TensorIterator& iter) {
+  AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES_AND1(ScalarType::Half, iter.dtype(), "cos_cuda", [&]() {
```

Comment:
You missed BFloat16

Reply:
Have added it. However do not that `sin_kernel_cuda` currently only dispatches to `Half` and not to `BFloat16`.